include/zephyr/input/input.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_INPUT_H_
+#define ZEPHYR_INCLUDE_INPUT_H_
+
+/**
+ * @brief Input Interface
+ * @defgroup input_interface Input Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <stdint.h>
+#include <zephyr/device.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Input event structure.
+ *
+ * This structure represents a single input event, for example a key or button
+ * press for a single button, or an absolute or relative coordinate for a
+ * single axis.
+ */
+struct input_event {
+	/** Device generating the event or NULL. */
+	const struct device *dev;
+	/** Event type (see @ref INPUT_EV_CODES). */
+	uint16_t type;
```

Comment:
It is weird to exchange the sync state inside the type field without explicitly saying that. I would use some other approach to solve that, like:
```suggestion
union {
        struct {
                /** Event sync. */
                uint16_t sync: 1;
                /** Event type (see @ref INPUT_EV_CODES). */
                uint16_t type: 15;
        };
        uint16_t sync_type;
};

```

Reply:
> exchange the sync state inside the type field without explicitly saying that

It's literally one of `INPUT_EV_CODES`. I see how that make this harder to use for a message system, no need to word around it.

Made some space for it, not keen on bit fields outside of protocol code so gave it a `uint8_t`. It departs from the Linux structure but I guess it was not following that already anyway.